### DISCUSSION TITLE
How to get the sizes of the rings that the specific bond is in?

### DISCUSSION THREAD

**Discussion by eereenah-fast** [rdkit/rdkit#4517] [Q&A] [answered]
Hello! 

What is the fastest/easiest way to get the sizes of the rings that the specific bond is in? I found the definition of [`bondRindSizes()`](https://www.rdkit.org/docs/cppapi/classRDKit_1_1RingInfo.html#ac695515b223ee5586ff35476c196fa4d) but I am not sure how to use this in Python.

Thanks!

**Comment by ptosco** [comment, marked as answer]
@eereenah AFAICT `bondRingSizes()` and `atomRingSizes()` are not exposed to Python. I will create an issue for that as it would be handy if they were.
For the time being I'd use a list comprehension:
```
from rdkit import Chem
from rdkit.Chem.Draw import IPythonConsole

IPythonConsole.drawOptions.addBondIndices = True

mol = Chem.MolFromSmiles("C12CCCC1CCCC2")
```
![image](https://user-images.githubusercontent.com/5244385/133430408-c5da6856-f881-48d2-a6f7-451aa02cfdf4.png)
```
ri = mol.GetRingInfo()
for b in mol.GetBonds():
    print(b.GetIdx(), [len(ring) for ring in ri.BondRings() if b.GetIdx() in ring])
0 [5]
1 [5]
2 [5]
3 [5]
4 [6]
5 [6]
6 [6]
7 [6]
8 [5, 6]
9 [6]
```

**Reply by eereenah-fast** [reply]
Thanks a lot!
